File: sdk/src/main/java/org/maibot/sdk/SdkVersion.java
package org.maibot.sdk;

import java.io.InputStream;
import java.util.Properties;

public class SdkVersion {
    public static String get() {
        try (InputStream input = SdkVersion.class.getResourceAsStream("/META-INF/build-inf.properties")) {
            assert input != null;

            Properties prop = new Properties();
            prop.load(input);

            return prop.getProperty("version", "0.0.0");
        } catch (Exception e) {
            throw new RuntimeException("An error occurred when loading version info.", e);
        }
    }
}

File: sdk/src/main/java/org/maibot/sdk/SdkVersion.kt
package org.maibot.sdk

import org.maibot.sdk.exceptions.FatalError
import java.io.IOException
import java.util.*

class SdkVersion {
    companion object {
        fun get(): String {
            try {
                SdkVersion::class.java.getResourceAsStream("/org/maibot/sdk/build-inf.properties").use { input ->
                    checkNotNull(input)
                    val prop = Properties()
                    prop.load(input)
                    return prop.getProperty("version", "0.0.0")
                }
            } catch (e: IOException) {
                throw FatalError("An error occurred when loading version info.", e)
            }
        }
    }
}
